### PR TITLE
Add a PhoneNumber class to handle formatting and actions on numbers

### PR DESCRIPTION
**Pull request by caitlinrhd**
We value collaboration at Invoca, and focus on peer mentorship as a way for the whole team to learn and thrive. As part of this process, we conduct code reviews during our development cycle. In this phase of the interview, we would like you to review and comment on this Pull Request. You can assume that this Pull Request was written by an intern, who is new to professional software development.  Please take about 20-30 minutes to review this Pull Request and comment as if you were reviewing an intern's code.

### CODE REVIEW

File: phone_actions.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class PhoneActions
+
+  def initialize(username, password)
+    @actions_url = "https://#{username}:#{password}@api.<made_up>.com/actions/"
+  end
+
+  def call!(number, message)
+    JSON.parse(RestClient.post("@actions_url/#{text}", {message: message, number: number}))
```
[the remark this answers is not in the document]
Comment (jmsv6d) [reply phone_actions.rb:8]: How will api error responses be handled?

File: phone_number.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  def format_number
+    "(#{find_area_code}) #{find_exchange}-#{find_line_number}"
+  end
+
+  def find_area_code
+    open_paren = @number.index('(')
+    close_paren = @number.index(')')
+    if open_paren
+      if close_paren
```
[the remark this answers is not in the document]
Comment (jmsv6d) [reply phone_number.rb:44]: Other conditions to consider would be if an incorrect number is received. Could there be more than 9 digits passed in? Less? With this regex any extra numbers would be cut off at the end, is that how this case should be handled?

File: phone_number_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+require_relative "./phone_number"
+
+RSpec.describe PhoneNumber do
```
Comment (jmsv6d) [inline phone_number_spec.rb:3]: Make sure to test for the edge cases I mentioned in previous comments (invalid numbers, error responses, invalid phone number formats, etc)
